src/lib/operators/sort.cpp

```diff
@@ -33,19 +20,78 @@ const std::string& Sort::name() const {
 std::shared_ptr<AbstractOperator> Sort::_on_deep_copy(
     const std::shared_ptr<AbstractOperator>& copied_input_left,
     const std::shared_ptr<AbstractOperator>& copied_input_right) const {
-  return std::make_shared<Sort>(copied_input_left, _column_id, _order_by_mode, _output_chunk_size);
+  return std::make_shared<Sort>(copied_input_left, _sort_definitions, _output_chunk_size);
 }
 
 void Sort::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {}
 
 std::shared_ptr<const Table> Sort::_on_execute() {
-  _impl = make_unique_by_data_type<AbstractReadOnlyOperatorImpl, SortImpl>(
-      input_table_left()->column_data_type(_column_id), input_table_left(), _column_id, _order_by_mode,
-      _output_chunk_size);
-  return _impl->_on_execute();
+  _sort_definition_data_types.resize(_sort_definitions.size());
+  for (auto column_id = ColumnID{0}; column_id < _sort_definitions.size(); ++column_id) {
+    auto sort_definition = _sort_definitions[column_id];
+    _sort_definition_data_types[column_id] = input_table_left()->column_data_type(sort_definition.column);
+  }
+
+  std::shared_ptr<const Table> output = input_table_left();
+  std::shared_ptr<PosList> previously_sorted_pos_list = std::shared_ptr<PosList>(nullptr);
+
+  _validate_sort_definitions();
+
+  for (auto column_id = _sort_definitions.size(); column_id-- != 0;) {
+    const bool is_last_sorting_run = (column_id == 0);
+
+    auto sort_definition = _sort_definitions[column_id];
+    DataType data_type = _sort_definition_data_types[column_id];
+
+    resolve_data_type(data_type, [&](auto type) {
+      using ColumnDataType = typename decltype(type)::type;
+
+      auto sort_single_col_impl = SortImpl<ColumnDataType>(input_table_left(), sort_definition.column,
+                                                           sort_definition.order_by_mode, _output_chunk_size);
+      const auto row_id_value_vector = sort_single_col_impl.sort_one_column(previously_sorted_pos_list);
+
+      // 3. Materialization of the result: We take the sorted ValueRowID Vector, create chunks fill them until they are
```

Comment:
I can't find points 1 and 2. Also needs a comma after create chunks.

Reply:
Please also use consistent naming between ValueRowID and row_id_value_vector.

Reply:
fixed in 32efd0affe63408e18085d1837f40091cb078bce